src/OmniSharp.LanguageServerProtocol/LanguageServerHost.cs

```diff
@@ -94,7 +94,7 @@ private void CreateCompositionHost(InitializeParams initializeParams)
             var compositionHostBuilder = new CompositionHostBuilder(_serviceProvider, _environment, eventEmitter)
                 .WithOmniSharpAssemblies()
                 .WithAssemblies(typeof(LanguageServerHost).Assembly)
-                .WithAssemblies(plugins.AssemblyNames.Select(Assembly.Load).ToArray());
+                .WithAssemblies(plugins.LoadAssemblies().ToArray());
```

Comment:
i think you could do `var assemblyLoader = _serviceProvider.GetRequiredService<IAssemblyLoader>();` here and use the OmniSharp assembly loader instead, this way the assembly loading logic would be always the same

Reply:
Thanks, I didn't notice that. 

`IAssemblyLoader` has different methods for `Load` (which in the one implementation calls `Assembly.Load`) and `LoadFrom` (which calls `Assembly.LoadFrom`) so the method for deciding which to call based on file existence still needs to be added somewhere. It could be added to `IAssemblyLoaderExtensions` but would need a clearer name (e.g. `LoadByAssemblyNameOrPath` would be clear but verbose.)

The question is, since this is rationalizing command-line arguments for the `-pl` option, maybe resolving this ambiguity belongs closer to the parsing operation anyway?

Reply:
@filipw Thoughts on the latest changes?

Reply:
sorry for the delay - what I actually meant is that this logic would ideally utilize the already existing methods of the loader (or its extension methods) - as they seem to cover everything needed.

 For example there is already a method to load an assembly by string name (here https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Abstractions/Services/IAssemblyLoader.cs#L23), by multiple names (here https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Abstractions/Services/IAssemblyLoader.cs#L34) or to load it from a given path (here https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Host/Services/AssemblyLoader.cs#L64). So you don't need to add any new ones, you just need to make the decision if it's a file or not a file and call the relevant existing method.

The advantage of this is also that the existing methods handle exceptions so they would prevent the logic from crashing already at startup and they log so we could easily see in the server logs which assemblies have been successfully loaded as plugins.